src/psyclone/tests/dependency_test.py

```diff
@@ -0,0 +1,329 @@
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: J. Henrichs, Bureau of Meteorology
+
+
+''' Module containing py.test tests for dependency analysis.'''
+
+from __future__ import print_function, absolute_import
+import os
+import pytest
+
+from fparser.common.readfortran import FortranStringReader
+from psyclone import nemo
+from psyclone.core.access_info import VariablesAccessInfo
+from psyclone.core.access_type import AccessType
+from psyclone.psyGen import Assignment, IfBlock, Loop, PSyFactory
+
+
+# Constants
+API = "nemo"
+# Location of the Fortran files associated with these tests
+BASE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "test_files")
+
+
+def test_assignment(parser):
+    ''' Check that assignments set the right read/write accesses.
+    '''
+    reader = FortranStringReader('''program test_prog
+                                 a = b
+                                 c(i,j) = d(i,j+1)+e+f(x,y)
+                                 c(i) = c(i) + 1
+                                 d(i,j) = sqrt(e(i,j))
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    # Simple scalar assignment:  a = b
+    scalar_assignment = schedule.children[0]
+    assert isinstance(scalar_assignment, Assignment)
+    var_accesses = VariablesAccessInfo()
+    scalar_assignment.reference_accesses(var_accesses)
+    # Test some test functions explicitly:
+    assert var_accesses.is_written("a")
+    assert not var_accesses.is_read("a")
+    assert not var_accesses.is_written("b")
+    assert var_accesses.is_read("b")
+
+    # Array element assignment: c(i,j) = d(i,j+1)+e+f(x,y)
+    array_assignment = schedule.children[1]
+    assert isinstance(array_assignment, Assignment)
+    var_accesses = VariablesAccessInfo()
+    array_assignment.reference_accesses(var_accesses)
+    assert str(var_accesses) == "c: WRITE, d: READ, e: READ, f: READ, "\
+                                "i: READ, j: READ, x: READ, y: READ"
+
+    # Increment operation: c(i) = c(i)+1
+    increment_access = schedule.children[2]
+    assert isinstance(increment_access, Assignment)
+    var_accesses = VariablesAccessInfo()
+    increment_access.reference_accesses(var_accesses)
+    assert str(var_accesses) == "c: READWRITE, i: READ"
+
+    # Using an intrinsic (looks like an array access): d(i, j) = sqrt(e(i, j))
```

Comment:
Not any more! (Sorry.) Please bring this branch up-to-date and fix...

Reply:
Luckily it works out of the box for now, it just sets e as read. The one issue that we might need to solve if we have intrinsics that modify a parameters.